src/code.c

```diff
@@ -259,6 +259,10 @@ Stat NewStatOrExpr(CodeState * cs, UInt type, UInt size, UInt line)
     StatHeader * header = STAT_HEADER(cs, stat);
     header->line = line;
     header->size = size;
+    // check size fits inside header
+    if (header->size != size) {
```

Comment:
Unfortunately this is not enough in the 32bit case...

Reply:
I think this was OK for 32-bits, just the test needing fixing (as 32-bit objects are smaller, we fit more in the allowed size)

Reply:
makes sense!